cypress/e2e/smoke/address_book.cy.js

```diff
@@ -20,91 +20,94 @@ const GNO_CSV_ENTRY = {
-      // Import CSV
-      cy.findByText('Import').click()
+      cy.contains('Import').click()
       cy.get('[type="file"]').attachFile('../fixtures/address_book_test.csv')
-      //This is the submit button for the Import modal. It requires an actuall class or testId to differentiate
-      //from the Import button at the top of the AB table
-      cy.contains('button.MuiButton-sizeMedium', 'Import').click()
-      cy.findByText(RINKEBY_CSV_ENTRY.name).should('exist')
-      cy.findByText(RINKEBY_CSV_ENTRY.address).should('exist')
-    })
 
-    it.skip('should find Gnosis Chain imported address', () => {
-      //Test skipped because it causes an error in the simulation
-      //"Unhandled Runtime Error
-      //Error: Hydration failed because the initial UI does not match what was rendered on the server."
+      // Import button should be enabled
+      cy.get('.MuiDialogActions-root').contains('Import').should('not.be.disabled')
+      cy.get('.MuiDialogActions-root').contains('Import').click()
 
+      // The import modal should be closed
+      cy.get('Import address book').should('not.exist')
+      cy.contains(RINKEBY_CSV_ENTRY.name).should('exist')
+      cy.contains(RINKEBY_CSV_ENTRY.address).should('exist')
+    })
+
+    it('should find Gnosis Chain imported address', () => {
       // Go to a Safe on Gnosis Chain
-      cy.get('[aria-haspopup="listbox"]').findByText('Rinkeby').click()
-      cy.findByText('Gnosis Chain').click()
+      cy.get('header')
+        .contains(/^Rinkeby$/)
+        .click()
+      cy.contains('Gnosis Chain').click()
 
       // Navigate to the Address Book page
-      cy.visit(`/${GNO_TEST_SAFE}/address-book`)
+      cy.visit(`/${GNO_TEST_SAFE}/address-book`, { failOnStatusCode: false })
 
-      // Close cookies banner
-      cy.contains('button', 'Accept selection').click()
       // Waits for the Address Book table to be in the page
       cy.contains('p', 'Address book').should('be.visible')
 
       // Finds the imported Gnosis Chain address
-      cy.findByText(GNO_CSV_ENTRY.name).should('exist')
-      cy.findByText(GNO_CSV_ENTRY.address).should('exist')
+      cy.contains(GNO_CSV_ENTRY.name).should('exist')
+      cy.contains(GNO_CSV_ENTRY.address).should('exist')
     })
 
     it('should download correctly the address book file', () => {
       // Download the export file
       const date = format(new Date(), 'yyyy-MM-dd')
       const fileName = `safe-address-book-${date}.csv` //name that is given to the file automatically
 
-      cy.contains('button', 'Export').click()
+      // safe-address-book-2022-09-21
+
+      cy.contains('Export').click()
       //This is the submit button for the Export modal. It requires an actuall class or testId to differentiate
       //from the Export button at the top of the AB table
-      cy.contains('button.MuiButton-sizeMedium', 'Export').click()
+      cy.get('.MuiDialogActions-root').contains('Export').click()
```

Comment:
Same for this one.

Reply:
There is another button with 'Export' text thus going more specific